ibis/backends/pandas/execution/window.py

```diff
@@ -501,17 +501,24 @@ def execute_series_group_by_last_value(op, data, aggcontext=None, **kwargs):
 
 @execute_node.register(ops.MinRank, (pd.Series, SeriesGroupBy))
 def execute_series_min_rank(op, data, **kwargs):
-    # TODO(phillipc): Handle ORDER BY
     return data.rank(method='min', ascending=True).astype('int64') - 1
 
 
 @execute_node.register(ops.DenseRank, (pd.Series, SeriesGroupBy))
 def execute_series_dense_rank(op, data, **kwargs):
-    # TODO(phillipc): Handle ORDER BY
     return data.rank(method='dense', ascending=True).astype('int64') - 1
 
 
-@execute_node.register(ops.PercentRank, (pd.Series, SeriesGroupBy))
+@execute_node.register(ops.PercentRank, pd.Series)
 def execute_series_percent_rank(op, data, **kwargs):
-    # TODO(phillipc): Handle ORDER BY
+    return (data.rank(method='min') - 1) / (len(data) - 1)
+
+
+@execute_node.register(ops.PercentRank, SeriesGroupBy)
+def execute_series_group_by_percent_rank(op, data, **kwargs):
+    return (data.rank(method='min') - 1) / (data.transform(len) - 1)
+
+
+@execute_node.register(ops.CumeDist, (pd.Series, SeriesGroupBy))
+def execute_series_cume_dist(op, data, **kwargs):
```

Comment:
are we actually testing this with hard values? how do you know this is correct

Reply:
not sure if I understood the question. But the tests for cume_dist and percent_rank are here: https://github.com/ibis-project/ibis/pull/2224/files#diff-4dffd4905e524c28c8eaf89eed6633b319db9564c5ade224fe775bffad718b82R57-R66

do you want me to add more tests for that? if positive, could you provide more details about the tests you want?

Reply:
@jreback are the tests I pointed above enough? if it is not, let me know what do you expect for these tests pls. thanks

Reply:
The tests seem consistent with the previous ones. I agree that it would make sense to test with actual values (e.g. `1, 2, 3,...`) instead of comparing the result of the function with the values generated by the `rank` function (the same being used).

But I think this should be changed for all the existing functionality ('lag', 'dense_rank'...). @xmnlab can you create a separate issue for that, with a link to the tests that need to be changed? Thanks!

Reply:
sure I will open the issue now. thanks!